### PR TITLE
golang filter extension: simple fuzz for downstream input.

### LINKED ISSUE #25807: golang filter extension: increase fuzz coverage
Seems the fuzz for the ext_proc filter could be a good example to follow up.
https://github.com/envoyproxy/envoy/pull/17636

key points:
1. random downstream request
2. random upstream response
3. random Go filter behavior: sync/async, sleep in Go, GetHeader/SetHeader, and other APIs.

In my opinion,

1 & 2 could be the standard fuzz way, but may not be very useful for golang filter, since golang filter reuses the existing Envoy C++ APIs mostly. But it is still good to have it.

3 might be tailored for golang filter, similar to the [existing basic test filter](https://github.com/envoyproxy/envoy/tree/main/contrib/golang/filters/http/test/test_data/basic).
That is, we have multiple test modes in the test Go filter, https://github.com/envoyproxy/envoy/blob/main/contrib/golang/filters/http/test/test_data/basic/filter.go#L22-L29
And we enable them randomly in the fuzz test. (we only enable some of the patterns in the base test, non-fuzz mode)

### PR DESCRIPTION
**Pull request by doujiang24**
Commit Message:
Additional Description:
Partial fix #25807

Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: contrib/golang/filters/http/test/golang_filter_fuzz.proto [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+package envoy.extensions.filters.http.golang;
+
+import "contrib/envoy/extensions/filters/http/golang/v3alpha/golang.proto";
+import "test/fuzz/common.proto";
+import "envoy/config/core/v3/base.proto";
+import "validate/validate.proto";
+
+// We only fuzz a single request per iteration.
+message GolangFilterTestCase {
+  envoy.extensions.filters.http.golang.v3alpha.Config config = 1
```
Comment (adisuissa) [inline contrib/golang/filters/http/test/golang_filter_fuzz.proto:11]: One thing to consider is what is the threat model that you'd like this fuzzer to address.
Filter config is typically considered trusted (coming from a trusted domain). Our previous experience shows that fuzzing the config raises false-errors - errors that should not actually occur.
This is ok, but reduces the ability of the fuzzer to detect data-plane issues.
Consider using pre-defined configs with the same `request_data`.

**Review by adisuissa — COMMENTED** [review]
Overall LGTM, thanks!
Left one issue that you should consider.

Please also let this fuzzer run for a few hours, just to see if it catches some errors that should be fixed before this goes live.

File: contrib/golang/filters/http/test/golang_filter_fuzz.proto (hunk 1 above)
Comment (doujiang24) [reply contrib/golang/filters/http/test/golang_filter_fuzz.proto]: Sorry for the late, busy with other stuff.

I already removed the `config` in this proto, and changed to use a const config for fuzzing.
I'm not sure I understand the `pre-defined` configs, please correct me if I'm wrong, thanks.

**Review by adisuissa — APPROVED** [review]
LGTM, thanks!

**Pull request merged**